product_attribute_model_link/models/product_attribute_value.py

```diff
@@ -8,10 +8,25 @@
 class ProductAttributeValue(models.Model):
     _inherit = "product.attribute.value"
 
+    model = fields.Char("Related Record Model")
+    res_id = fields.Many2oneReference("Related Record ID", model_field="model")
     linked_record_ref = fields.Reference(
-        selection="_selection_target_model", string="Linked record"
+        selection="_selection_target_model",
+        string="Linked record",
+        compute="_compute_record_ref",
     )
 
+    @api.depends("model", "res_id")
+    def _compute_record_ref(self):
+        for rec in self:
+            if rec.model and rec.res_id:
+                res = self.env[rec.model].browse(rec.res_id)
+                result = res.check_access_rights("read", raise_exception=False)
```

Comment:
`result` name looks confusing for such function result. Better use something more explicit. Eg `can_read`.